tests/functional/behave_features/steps/implementation.py

```diff
@@ -198,4 +209,41 @@ def the_user_creates_a_branch_to_add_a_new_chart_version(context):
 def chart_yaml_specifies_bad_version(context, bad_version):
     if bad_version != '':
         context.workflow_test.update_bad_version(bad_version)
-        
\ No newline at end of file
+
+@given(u'the report contains "{error}"')
+def sha_value_does_not_match(context, error):
+    if error == 'sha_mismatch':
+        context.workflow_test.process_report(update_chart_sha=True)
+    else:
+        raise AssertionError(f"This {error} handling is not implemented yet")
+
+@when(u'the user sends a pull request with the chart tar and report')
+def user_sends_pull_request_with_chart_tarball_and_report(context):
+    context.workflow_test.push_chart(is_tarball=True)
+    context.workflow_test.send_pull_request()
+
+######## Test Submitted Charts Step definitions ##########
+@given(u'there is a github workflow for testing existing charts')
+def theres_github_workflow_for_testing_charts(context):
+    print("[INFO] Running step: there is a github workflow for testing existing charts")
+
+@when(u'a new Openshift or chart-verifier version is specified')
+def new_openshift_or_verifier_version_is_specified(context):
```

Comment:
confused by a step which only prints a message. If they do not actually do anything they should be deleted from the features. There are quite a few like this.

Reply:
test_submitted_chart was implemented in a single function process_all_charts() . Since feature file contains multiple given, when and then but basically do nothing. One way would be break the process_all_charts(), that requires significant changes to the code. I think that requires a story, we can discuss more. @mmulholla  WDYT?

Reply:
OK, that feature file is totally flawed. We can leave t for now and when we fix the workflow we can update the feature too. Real problem is it is not user scenario.